internal/store/utils.go

```diff
@@ -175,6 +182,17 @@ func isPrefixedNativeResource(name v1.ResourceName) bool {
 	return strings.Contains(string(name), v1.ResourceDefaultNamespacePrefix)
 }
 
+// convenience wrapper to inject allow-listed labels and annotations to a metric if per-series injection is enabled.
+func injectLabelsAndAnnos(m *metric.Metric, metricConfig *MetricConfig, obj *metav1.ObjectMeta) *metric.Metric {
+	if !metricConfig.InjectPerSeriesMetadata {
```

Comment:
I _think_ with the pass-by-reference + early guard clause, this should be "0-cost" for those leaving the `--inject-per-series-metadata` at it's default of `false`.